src/TaskLayout.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Various rendering options for a query.
+ */
+export class LayoutOptions {
+    hideTaskCount: boolean = false;
+    hideBacklinks: boolean = false;
+    hidePriority: boolean = false;
+    hideStartDate: boolean = false;
+    hideScheduledDate: boolean = false;
+    hideDoneDate: boolean = false;
+    hideDueDate: boolean = false;
+    hideRecurrenceRule: boolean = false;
+    hideEditButton: boolean = false;
+    hideUrgency: boolean = true;
+    shortMode: boolean = false;
+}
+
+export type LayoutComponent =
+    | 'description'
+    | 'priority'
+    | 'recurrenceRule'
+    | 'startDate'
+    | 'scheduledDate'
+    | 'dueDate'
+    | 'doneDate'
+    | 'blockLink';
+
```

Comment:
I was confused as to `whyLayoutComponent` did not include all the values in LayoutOptions.

From reading the code, I think the answer is probably that that it only contains
the options that reflect the contents of a rendered Task.

If that is correct, I think things would be clearer if this was named something like `TaskLayoutComponent`

Reply:
Will do. You understood correctly and `TaskLayoutComponent` is indeed better.